### PR TITLE
Support for ConductR

### PR DESCRIPTION
**Pull request by huntc**
Current progress on allowing RM to be deployed to ConductR.

To run:

(from within sbt):

```
bundle:dist
backend-region:dist
backend-summary:dist

set SandboxKeys.image in Global := "conductr/conductr"
set SandboxKeys.imageVersion in Global := "latest"
sandbox run --withFeatures logging
```

...then load and run the bundles (*substitute your paths and hashes as appropriate*):

> Note that the following presently requires a local build of sbt-conductr - be sure to update the `plugins.sbt` to a locally built one of https://github.com/sbt/sbt-conductr (we will be releasing a new version based on this shortly).

```
conduct load /pathtotheproject/target/bundle/reactive-maps-frontend-v1-76d4af00c95dbed5eddc4a7a93a753bf7a85c14771745eea79239b52f61b90a0.zip target/bundle-configuration/backend-region-088d9a0e75c73eedf5e9f492cb5e5ec66f3a04fd8b7cbb64cf2b1e833691144a.zip
conduct load /pathtotheproject/target/bundle/reactive-maps-backend-region-v1-76d4af00c95dbed5eddc4a7a93a753bf7a85c14771745eea79239b52f61b90a0.zip target/bundle-configuration/backend-summary-52e1a1dacd377993aec9c3ac81fc69242f5518a3e09abe4fac22887a2a3b59c0.zip
conduct load /pathtotheproject/target/bundle/reactive-maps-backend-summary-v1-76d4af00c95dbed5eddc4a7a93a753bf7a85c14771745eea79239b52f61b90a0.zip target/bundle-configuration/frontend-03202dc543f98519752a76b7581621687ec7eb401a7b67bbe26cf9202a70f8fe.zip

conduct run reactive-maps-frontend
conduct run reactive-maps-backend-region
conduct run reactive-maps-backend-summary
```

You should ultimately see something like this:

```
ID               NAME                           #REP  #STR  #RUN
76d4af0-03202dc  reactive-maps-frontend            1     0     1
23b4342          conductr-elasticsearch            1     0     1
60dd1b7          conductr-kibana                   1     0     1
76d4af0-088d9a0  reactive-maps-backend-region      1     0     1
76d4af0-52e1a1d  reactive-maps-backend-summary     1     0     1
```

TODO: Also update the tutorial and explain how to do the above.

### CODE REVIEW

**Comment by huntc** [comment]
@jroper can you shed light on why the PR is failing? The tests actually pass.

**Comment by jroper** [comment]
The tutorial is broken.  See the README here for how the Activator tutorial works to generate and validate code embedded in tutorials:

https://github.com/typesafehub/activator-tutorial-generator

In this case, the problem most likely is that the offsets in the edit scripts have changed since `Main.scala` has changed, so the edit scripts are being applied in the wrong location of the file.  They will need to be updated - this is unfortunately one of the more tedious things about maintaining the validated tutorials.  If we used full context diffs we probably wouldn't have this problem, but then full context diffs are much harder to maintain by hand than edit scripts, which are very easy for a human to read and write.

**Comment by huntc** [comment]
@jroper @rkuhn @markusjura Can I please get a review - don't merge just yet as I require some plugin updates still (Markus is doing that today for me). Reviewing the tutorial would be particularly useful to me. 

Thanks!

**Comment by huntc** [comment]
@pvlugter Could you also please review, particularly by following the steps in the tutorial...?

**Comment by ktoso** [comment]
Read through the PR, looks ok. Not much to review here akka-wise though, let me know if there's something specific you wanted to confirm here @huntc (PubSub guarantees you state are correct).

**Comment by huntc** [comment]
Thanks @ktoso! 

@markusjura?

File: tutorial/index.html.script [hunk 1]
```diff
@@ -1512,6 +1538,120 @@ object SharedJournalHelper {
...
+    <p>
+      sbt-conductr actually brings in another plugin named <a href="https://github.com/sbt/sbt-bundle" target="_blank">sbt-bundle</a> and between them, you declare some
+      <a href="#code/build.sbt" class="shortcut">settings</a> required for deploying your application.
+      You then load and run a "bundle" which is the
+      unit of deployment for ConductR. A bundle is a zip file with the hash of its contents encoded in its name.
+      Think of the hash as a verifiable version number. When the zip file's contents changes then the
+      hash will change as well. ConductR verifies this hash thus providing the confidence that version xyz
+      is absolutely version xyz and has not been tampered with in any way. Given the uniqueness of this hash, you
+      can support rolling upgrades by having multiple versions of your application running at the same time. You
+      also know what point to rollback to should an upgrade of your application not go the way you planned.
```
Comment (jroper) [inline tutorial/index.html.script]: This feels to me like too much detail for the reactive maps tutorial while at the same time not being enough detail to adequately explain the importance and motivations behind bundle hashes.  Perhaps just mentioning that a bundle is identified by its hash, and a link to the ConductR documentation on bundle hashes would be better?

Comment (huntc) [inline tutorial/index.html.script]: OK

**Comment by markusjura** [comment]
LGTM

**Comment by huntc** [comment]
3 people reviewed. I've also gone through the tutorial, ran the standalone, and then the multi-jvm, and then the ConductR approach. All is looking good from here. I'll be merging this PR once the validation has complete.

**Pull request merged**